packages/headless/src/controllers/commerce/sort/core/headless-core-commerce-sort.test.ts

```diff
@@ -0,0 +1,98 @@
+import {Action} from '@reduxjs/toolkit';
+import {fetchProductListing} from '../../../../features/commerce/product-listing/product-listing-actions';
+import {applySort} from '../../../../features/commerce/sort/sort-actions';
+import {sortReducer} from '../../../../features/commerce/sort/sort-slice';
+import {updatePage} from '../../../../features/pagination/pagination-actions';
+import {buildMockCommerceEngine, MockCommerceEngine} from '../../../../test';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  buildRelevanceSortCriterion,
+  buildCoreSort,
+  Sort,
+  SortBy,
+} from './headless-core-commerce-sort';
+
+describe('headless core sort', () => {
+  let sort: Sort;
+  let engine: MockCommerceEngine;
+  const fetchResultsActionCreator = fetchProductListing;
+
+  beforeEach(() => {
+    engine = buildMockCommerceEngine();
+    sort = buildCoreSort(engine, {
+      fetchResultsActionCreator,
+    });
+  });
+
+  const expectContainAction = (action: Action) => {
+    const found = engine.actions.find((a) => a.type === action.type);
+    expect(engine.actions).toContainEqual(found);
+  };
+
+  it('adds the correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      commerceSort: sortReducer,
+    });
+  });
+
+  it('dispatches #applySort on load when sort is specified', () => {
+    sort = buildCoreSort(engine, {
+      initialState: {
+        criterion: buildRelevanceSortCriterion(),
+      },
+      fetchResultsActionCreator,
+    });
+    expectContainAction(applySort);
+  });
+
+  it('sortBy dispatches #applySort, #updatePage, and #fetchResultsActionCreator', () => {
+    sort.sortBy(buildRelevanceSortCriterion());
+    expectContainAction(applySort);
+    expectContainAction(updatePage);
```

Comment:
If you agree with my suggestion of handling pagination reset in the pagination reducer, you should remove this.